dashboard/lib/services/user.rb

```diff
@@ -3,6 +3,7 @@ class Services::User
   # new values to the given User instance.
   def self.assign_form_params(user, params)
     assign_auth_option_params(user, params)
+    assign_school_info_params(user, params)
     # Remove nested attributes, otherwise we might accidentally create new
     # nested objects rather than updating existing ones.
     # '_attributes' is how active_record identifies nested attrs.
```

Comment:
Maybe a question for the future but why is removing all nested attributes necessary? If there was a specific attribute that was causing problems, could we handle that one separately and allow Rails to handle the rest?

Reply:
I honestly don't remember - I think it might have also worked to handle auth options in a special way and leave the rest. The change was specifically around handling auth options, because we wanted to be able to set the email address if it was missing in the LTI response. The idea was to ensure that we are updating the email in the existing auth_option instead of creating another one with the user-input email.

Reply:
@daynew do you remember why we didn't just reject `authentication_options_attributes` and let other nested attributes behave normally?

Reply:
I'd like to +1 Bethany's comment above. It feels more future-proof and true to Rails to handle auth as a special case but maintain the original nested attribute processing. That said, if it's a tougher change then I understand it just being a potential follow up.

Reply:
Ok I refactored based on this feedback. Still seems to work - lmk what you think. Still want to wait and see if @daynew knows of a reason for rejecting nested attributes other than auth options before merging anything though.

Reply:
At the time I was leaning towards being more explicit about how nested attributes were handled. In retrospect, I think the code should look like how it is in this PR.